test-harness/specs/parameters-ac4.oas2.txt

```diff
@@ -0,0 +1,28 @@
+====test====
+When I send a request to an operation
+And the operation has a required header
+And in the request I sent that header is missing
+Then I should get a 400 response back
+And the payload of the response should be in application/problem+json content-type
+====spec====
+swagger: '2.0'
+paths:
+  /pet:
+    get:
+      parameters:
+        - in: header
+          name: x-age
+          required: true
+          type: number
+      responses:
+        200:
+          description: ok
+          schema:
+            type: string
+====server====
+mock -p 4010
+====command====
+curl -sIXGET http://127.0.0.1:4010/pet
+====expect====
+HTTP/1.1 400 Bad Request
```

Comment:
@XVincentX, seems like 400 to me since the requests was not syntactically correct (it was missing the required header). So it was lacking in the format required for the request. But I'm not so sure. Basically, for the status codes in these tests I followed what was established in ACs (and what I also thought were the correct ones to use). Having said that, If you believe it should be 412 I will change it, since I don't have firm belief it should be 400/422

Reply:
Yeah fair enough, I'm fine with 400 as well, I was just wondering. @philsturgeon — got any thoughts on this?

Reply:
@XVincentX i think this is a misunderstanding of what 412 is about. Precondition failed means something like, for example, you tried to use a `If-Unmodified-Since` or `If-None-Match` on a POST, and the server told you to get lost because that's daft.